### ISSUE TITLE
'| yaml' ?

### ISSUE THREAD

**Issue by gf-mse** [rcoh/angle-grinder#70]
Just discovered this project ( which is an awesome idea, btw ); I wonder if it supports `| yaml` -- and if not, then how hard it would be to add that ?

**Comment by rcoh** [comment]
It should be medium easy to add -- medium because it requires some mild restructuring to the rendering pipeline that's a bit more complex than adding a regular operator. We'd need to choose if `| yaml` would select a different render or if it would add some sort of `renderhint` to the rows -- I'm leaning towards the previous one I think. 

`| json` would also be a nice addition too...

I also need to think about whether a `--output=yaml` is the right answer as well

**Comment by gf-mse** [comment]
My sincere apologies -- I meant a YAML parser. ( YAML output probably comes as a very natural next step, though )

**Comment by rcoh** [comment]
After a bit more thought, this may be quite tricky. Currently, things operate on a line-by-line basis. You'd need to somehow flatten your YAML structure onto a single line or significantly refactor that way input works.

**Comment by gf-mse** [comment]
So for `json` input, everything is flattened to one line first? That is, we need a plugin that converts `yaml` input to a sequence of `json`-formatted lines .. what I suppose could be an external program. Ok, I get it. Thank you.

**Comment by rcoh** [comment]
`json` input assumes that the entire JSON record is on one line (which is the case for most logging applications)

**Comment by rcoh** [comment]
@gf-mse are your yaml documents separated by `---` or something similar? It would be easy to allow changing the document separator from `\n` to another string like that.

**Comment by gf-mse** [comment]
No, but it could be achieved. Worse comes to worse, I can pre-parse it and add a `---` on every de-indentation, or next top-level list/dictionary Item. 

What actually makes me think if I shall open another issue and ask whether there are plans for some plugin mechanism )

**Comment by rcoh** [comment]
I doubt I'd add a plugin mechanism -- the CLI is such that it's generally
better to `|` from one program to another to handle more specific task. I'm
considering handling indentation grouping of records, but it adds
considerable complexity to the ingest pipeline which is currently very
simple
